exporters/prometheus/exporter.go

```diff
@@ -34,38 +37,50 @@ import (
 type Exporter struct {
 	metric.Reader
 	Collector prometheus.Collector
-}
 
-// collector is used to implement prometheus.Collector.
-type collector struct {
-	metric.Reader
+	handler http.Handler
 }
 
-// config is added here to allow for options expansion in the future.
-type config struct{}
+var _ metric.Reader = &Exporter{}
 
-// Option may be used in the future to apply options to a Prometheus Exporter config.
-type Option interface {
-	apply(config) config
+// collector is used to implement prometheus.Collector.
+type collector struct {
+	reader metric.Reader
 }
 
 // New returns a Prometheus Exporter.
-func New(_ ...Option) Exporter {
-	// this assumes that the default temporality selector will always return cumulative.
-	// we only support cumulative temporality, so building our own reader enforces this.
+func New(opts ...Option) (*Exporter, error) {
+	cfg := newConfig(opts...)
+
+	// TODO (#????): Enable some way to configure the reader, but not change temporality.
 	reader := metric.NewManualReader()
-	e := Exporter{
-		Reader: reader,
-		Collector: &collector{
-			Reader: reader,
-		},
+
+	handler := promhttp.HandlerFor(cfg.gatherer, promhttp.HandlerOpts{})
+	collector := &collector{
+		reader: reader,
 	}
-	return e
+
+	if err := cfg.registerer.Register(collector); err != nil {
+		return nil, fmt.Errorf("cannot register the collector: %w", err)
+	}
+
+	e := &Exporter{
+		Reader:    reader,
+		Collector: collector,
+
+		handler: handler,
+	}
+
+	return e, nil
+}
+
+func (e *Exporter) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
This method is exported, it should have a comment.

```suggestion
ServeHTTP serves this Exporter's collected metrics in Prometheus exposition format.
func (e *Exporter) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```